meorg_client/client.py

```diff
@@ -60,27 +72,35 @@ def _make_request(self, method: str, endpoint: str, data: dict=None, headers: di
         method = method.upper()
 
         if method not in VALID_METHODS:
-            raise Exception(f'Invalid method {method}')
+            raise Exception(f"Invalid method {method}")
 
         # GET/PUT requests have the data interpolated into the url
-        if method in ['GET', 'PUT']:
-            endpoint = endpoint.format(data)
+        if method in ["GET", "PUT"]:
+            endpoint = endpoint.format(**data)
 
-        url = f"{self.base_url}/{endpoint}"
+        url = urljoin(self.base_url, endpoint)
         all_headers = {**self.headers, **headers} if headers else self.headers
-        response = requests.request(method, url, data=data, headers=all_headers)
 
-        if response.status_code == 200:
+        # Update the client with the last response
+        self.last_response = requests.request(
+            method, url, data=data, headers=all_headers
+        )
 
+        if self.last_response.status_code in [200, 202]:
```

Comment:
Move status code as separate const in another file and use those const at all the places

Reply:
Done, I'm now checking the range for success as per RFC 2616, rather than maintaining a list of HTTP status codes.